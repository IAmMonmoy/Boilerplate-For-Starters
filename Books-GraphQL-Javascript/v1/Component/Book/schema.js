import { gql } from 'apollo-server-express';

const schema = gql`
    type Query {
        books: [Book]
        book(id: String): [Book]
    }

    type Book {
        bookId:	String!
        title: String!
        year: String!
        description: String
    }`;

module.exports = { schema };