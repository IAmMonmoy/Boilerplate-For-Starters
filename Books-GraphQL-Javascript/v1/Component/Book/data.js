
const booksListWithDesciption = [
	{ bookId: '32e959aa-bb2d-11eb-8529-0242ac130003', title: 'The Door in the Wall', year: 1989, description: 'A crippled boy in fourteenth-century England proves his courage and earns recognition from the King.' },
	{ bookId: '378bd73a-bb2d-11eb-8529-0242ac130003', title: 'Amos Fortune, Free Man', year: 1989, description: 'The life of the eighteenth-century African prince who, after being captured by slave traders, was brought to Massachusetts where he was a slave until he was able to buy his freedom at the age of sixty.' },
	{ bookId: '3d634b02-bb2d-11eb-8529-0242ac130003', title: 'The Hobbit, Or, There and Back Again', year: 1989, description: 'Bilbo Baggins, a respectable, well-to-do hobbit, lives comfortably in his hobbit-hole until the day the wandering wizard Gandalf chooses him to take part in an adventure from which he may never return..' },
	{ bookId: '4d634b02-bb2d-11eb-8529-0242ac130003', title: 'The Hobbit, Or, There and Back Again', year: 1989, description: 'Bilbo Baggins, a respectable, well-to-do hobbit, lives comfortably in his hobbit-hole until the day the wandering wizard Gandalf chooses him to take part in an adventure from which he may never return..' },
	{ bookId: '5d634b02-bb2d-11eb-8529-0242ac130003', title: 'The Hobbit, Or, There and Back Again', year: 1989, description: 'Bilbo Baggins, a respectable, well-to-do hobbit, lives comfortably in his hobbit-hole until the day the wandering wizard Gandalf chooses him to take part in an adventure from which he may never return..' },
];

const bookList = [
	{ bookId: '32e959aa-bb2d-11eb-8529-0242ac130003', title: 'The Door in the Wall', year: 1989 },
	{ bookId: '378bd73a-bb2d-11eb-8529-0242ac130003', title: 'Amos Fortune, Free Man', year: 1950 },
	{ bookId: '3d634b02-bb2d-11eb-8529-0242ac130003', title: 'The Hobbit, Or, There and Back Again', year: 2012 },
	{ bookId: '4d634b02-bb2d-11eb-8529-0242ac130003', title: 'The Hobbit, Or, There and Back Again', year: 1989 },
	{ bookId: '5d634b02-bb2d-11eb-8529-0242ac130003', title: 'The Hobbit, Or, There and Back Again', year: 1989 },
];

const getBooks = function () {
	return new Promise(async (resolve, reject) => {
		try {
			resolve(bookList);
		}
		catch (err) {
			reject(err);
		}
	});
};

const getBook = function (id) {
	return new Promise(async (resolve, reject) => {
		try {
			resolve(booksListWithDesciption.filter(books => books.bookId === id));
		}
		catch (err) {
			reject(err);
		}
	});
};

module.exports = { getBooks, getBook };

