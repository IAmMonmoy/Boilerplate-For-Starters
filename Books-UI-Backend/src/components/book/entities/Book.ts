import { IsNotEmpty, Validate, IsInt } from 'class-validator';
import BaseObject from '../../../globalEntities/BaseObject';
import { CustomDateCheck } from '../../../utils/customValidators';
import IBook from '../interfaces/IBook';

export default class Book implements IBook {
    private _bookId: string;
    private _title: string;
    private _year: number;
    private _description: string;

    public get bookId() {
        return this._bookId;
    }

    public get title() {
        return this._title;
    }

    public get year() {
        return this._year;
    }

    public get description() {
        return this._description;
    }

    constructor(bookId: string, title: string, year: number, description: string) {
        this._bookId = bookId;
        this._title = title;
        this._year = year;
        this._description = description;
    }
}
