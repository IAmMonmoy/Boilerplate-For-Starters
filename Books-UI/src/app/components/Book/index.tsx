import React from 'react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
// @ts-ignore
import GenricBookImage from './assets/book.jpeg';
import styled from 'styled-components/macro';

interface BookProps {
  id: string;
  title: string;
  year: string;
  details?: string;
  size?: string;
  onClick?: (id: string) => any;
}

const Book = ({
  id,
  title,
  year,
  details,
  size = 'sm',
  onClick,
}: BookProps) => {
  return (
    <Wrapper width={size}>
      <Card>
        <Card.Img variant="top" src={GenricBookImage} />
        <Card.Body>
          <Card.Title>{title}</Card.Title>
          <Card.Subtitle>{year}</Card.Subtitle>
          {details && (
            <Card.Text className="book--description">{details}</Card.Text>
          )}
          {onClick && (
            <ButtonWrapper>
              <Button onClick={() => onClick(id)} variant="primary">
                Details
              </Button>
            </ButtonWrapper>
          )}
        </Card.Body>
      </Card>
    </Wrapper>
  );
};

const Wrapper = styled.div<{ width: string }>`
  margin: 5px;
  width: ${props => (props.width === 'sm' ? '18em' : '50em')};
  display: table;
  margin: 0 auto;
`;

const ButtonWrapper = styled.div`
  margin: 2px;
`;
export default Book;
