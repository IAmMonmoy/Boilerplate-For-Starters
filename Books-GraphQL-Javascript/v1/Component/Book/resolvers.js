import { getBooks, getBook } from './data';

const resolvers = {
  Query: {
    books: (parent, { }) => {
      return getBooks()
        .then(function (result) {
          return result;
        })
        .catch(function (err) {
          throw new Error(err.message);
        });
    },
    book: (parent, { id }) => {
      return getBook(id)
        .then(function (result) {
          return result;
        })
        .catch(function (err) {
          throw new Error(err.message);
        });
    }
  }
};

module.exports = { resolvers }