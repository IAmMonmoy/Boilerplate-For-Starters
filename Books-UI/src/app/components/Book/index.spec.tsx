import * as React from 'react';
import { render } from '@testing-library/react';
import Book from './index';

describe('<Book/>', () => {
  it('should match the snapshot', () => {
    const book = render(
      <Book id="id" title="title" year="year" details="details" />,
    );
    expect(book.container.firstChild).toMatchSnapshot();
  });
});
