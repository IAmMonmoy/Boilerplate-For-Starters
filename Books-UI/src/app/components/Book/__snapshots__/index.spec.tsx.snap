// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book/> should match the snapshot 1`] = `
.c0 {
  margin: 5px;
  width: 18em;
  display: table;
  margin: 0 auto;
}

<div
  class="c0"
  width="sm"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="book.jpeg"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        title
      </div>
      <div
        class="card-subtitle h6"
      >
        year
      </div>
      <p
        class="book--description card-text"
      >
        details
      </p>
    </div>
  </div>
</div>
`;
